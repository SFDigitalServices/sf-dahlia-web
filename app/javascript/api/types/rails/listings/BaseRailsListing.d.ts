import RailsListingDescriptor from "./RailsListingDescriptor"

type BaseRailsListing = {
  reservedDescriptor?: Array<RailsListingDescriptor>
  prioritiesDescriptor?: Array<RailsListingDescriptor>
  listingID: string
  chartTypes?: unknown
  Id: string
  Tenure: string
  Name: string
  Application_Due_Date: string
  Lottery_Results_Date: string
  Reserved_community_minimum_age: number
  hasWaitlist: boolean
  Units_Available: number
  Building_URL?: string
  Building_Name: string
  Building_City?: string
  Building_State?: string
  Building_Street_Address?: string
  Building_Zip_Code?: string
  Publish_Lottery_Results: boolean
  Lottery_Status: string
  RecordTypeId: string
  imageURL?: string
}

export default BaseRailsListing
