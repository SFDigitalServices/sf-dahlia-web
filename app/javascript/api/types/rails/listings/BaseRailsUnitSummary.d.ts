interface BaseRailsUnitSummary {
  unitType: string
  totalUnits: number
  minSquareFt?: number
  minOccupancy: number
  maxSquareFt?: number
  maxOccupancy: number
  listingID: string
  availability: number
}

export default BaseRailsUnitSummary
