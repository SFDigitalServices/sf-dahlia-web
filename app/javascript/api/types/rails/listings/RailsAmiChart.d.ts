export type RailsAmiChartValue = {
  year: string
  percent: number
  numOfHousehold: number
  chartType: string
  amount: number
}

export type RailsAmiChart = {
  percent: string
  values: Array<RailsAmiChartValue>
  year: string
  chartType?: string
}

export type RailsAmiChartMetaData = {
  percent: number
  type: string
  year: number
}
