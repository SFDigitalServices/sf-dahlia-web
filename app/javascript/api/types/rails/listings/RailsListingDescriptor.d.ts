type RailsListingDescriptor = {
  listingId: string
  name: string
  numberOfUnits: number
  shortDescription?: string
}

export default RailsListingDescriptor
