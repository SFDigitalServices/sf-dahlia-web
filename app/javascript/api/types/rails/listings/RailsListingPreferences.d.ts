export type RailsListingPreference = {
  NRHPDistrict: string
  appTotal: number
  description: string
  listingId: string
  listingPreferenceID: string
  name: string
  order: number
  pdfUrl: string
  preferenceName: string
  preferenceProofRequirementDescription: string
  readMoreUrl: string
  requiresProof: boolean
  unitsAvailable: number
}
