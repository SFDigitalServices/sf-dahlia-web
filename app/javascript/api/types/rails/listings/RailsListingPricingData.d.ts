export type RailsListingPricingData = {
  unitType: string
  unitMaxAMI: number
  totalUnits: number
  minSquareFt: number
  minRentalMinincome: number
  minPriceWithParking: number
  minPriceWithoutParking: number
  minPercentIncome: number
  minOccupancy: number
  minMonthlyRent: number
  minHoaDuesWithParking: number
  minHoaDuesWithoutParking: number
  maxSquareFt: number
  maxRentalMinIncome: number
  maxPriceWithParking: number
  maxPriceWithoutParking: number
  maxPercentincome: number
  maxOccupancy: number
  maxMonthlyRent: number
  maxHoaDuesWithParking: number
  maxHoaDuesWithoutParking: number
  listingID: string
  availability: number
  absoluteMinIncome: number
  absoluteMaxIncome: number
}
