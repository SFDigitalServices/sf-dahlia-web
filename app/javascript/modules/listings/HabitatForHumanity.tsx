import React from "react"
import { Heading, t } from "@bloom-housing/ui-components"

import { ListingAddress } from "../../components/ListingAddress"

export const getHabitatContent = (listing, stackedDataFxn) => {
  const getHeader = (header: string) => {
    return (
      <div
        className={
          "font-sans font-semibold text-base uppercase text-gray-750 tracking-wider border-0 border-b pb-2 mb-2 mt-4"
        }
      >
        {header}
      </div>
    )
  }
  const getHabitatContentRow = (prefix: string, content: string) => {
    return (
      <div>
        <span className={"font-semibold"}>{prefix}</span> {content}
      </div>
    )
  }

  const stackedData = stackedDataFxn(listing)

  return (
    <>
      <Heading priority={2} style={"cardHeader"} className={"order-1"}>
        {listing.Name}
      </Heading>
      <Heading priority={3} style={"cardSubheader"} className={"order-2"}>
        <ListingAddress listing={listing} />
      </Heading>
      <hr className={"mb-2"} />
      <div className={"text-gray-750"}>
        <Heading priority={3} style={"tableHeader"}>
          {t("listings.availableUnits")}
        </Heading>
        {getHeader(t("t.units"))}
        {stackedData.map((row) =>
          getHabitatContentRow(
            `${row.unitType.cellText}:`,
            `${row.availability.cellText} ${t("t.available")}`
          )
        )}

        {getHeader(t("listings.habitat.payments"))}
        <div>{t("listings.habitat.payments.desc")}</div>
        <div className={"mt-4 border-0 border-b pb-2 mb-2"}>
          {t("listings.habitat.infoSession")}
        </div>
      </div>
    </>
  )
}
