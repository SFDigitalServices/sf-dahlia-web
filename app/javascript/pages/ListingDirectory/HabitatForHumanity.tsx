import React from "react"

import { t } from "@bloom-housing/ui-components"

export const getHabitatContent = (listing, stackedDataFxn) => {
  const getHeader = (header: string) => {
    return (
      <div
        className={
          "font-sans font-semibold text-base uppercase text-gray-750 tracking-wider border-0 border-b pb-2 mb-2 mt-4"
        }
      >
        {header}
      </div>
    )
  }
  const getHabitatContentRow = (prefix: string, content: string) => {
    return (
      <div>
        <span className={"font-semibold"}>{prefix}</span> {content}
      </div>
    )
  }

  const stackedData = stackedDataFxn(listing)

  return (
    <div className={"text-gray-750"}>
      <div className={"font-alt-sans font-semibold text-base mb-2 text-gray-900"}>
        {t("listings.availableUnits")}
      </div>
      {getHeader(t("t.units"))}
      {stackedData.map((row) =>
        getHabitatContentRow(
          `${row.unitType.cellText}:`,
          `${row.availability.cellText} ${t("t.available")}`
        )
      )}

      {getHeader(t("listings.habitat.payments"))}
      <div>{t("listings.habitat.payments.desc")}</div>
      <div className={"mt-4  border-0 border-b pb-2 mb-2"}>{t("listings.habitat.infoSession")}</div>
    </div>
  )
}
