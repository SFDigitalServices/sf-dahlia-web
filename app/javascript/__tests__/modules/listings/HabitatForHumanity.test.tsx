import renderer from "react-test-renderer"
import { getHabitatContent } from "../../../modules/listings/HabitatForHumanity"
import { habitatListing } from "../../data/RailsSaleListing/listing-sale-habitat"

describe("HabitatForHumanity", () => {
  it("renders HabitatForHumanity component", () => {
    const tree = renderer
      .create(
        getHabitatContent(habitatListing, () => {
          return [{ unitType: { cellText: "" }, availability: { cellText: "" } }]
        })
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
