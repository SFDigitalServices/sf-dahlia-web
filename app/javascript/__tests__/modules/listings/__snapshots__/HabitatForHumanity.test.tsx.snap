// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitatForHumanity renders HabitatForHumanity component 1`] = `
<DocumentFragment>
  <h2
    class="text__large-primary order-1"
  >
    <a
      href="/listings/a0W8H000000HI2uUAG"
    >
      TEST: 36 Amber Drive
    </a>
  </h2>
  <h3
    class="text__medium-normal order-2"
  >
    <span>
      36 Amber Drive, San Francisco, 
      <abbr
        class="no-underline"
        title="California"
      >
        CA
      </abbr>
       94131
    </span>
  </h3>
  <hr
    class="mb-2"
  />
  <div
    class="text-gray-750"
  >
    <h3
      class="text__small-weighted"
    >
      Available Units
    </h3>
    <div
      class="font-sans font-semibold text-base uppercase text-gray-750 tracking-wider border-0 border-b pb-2 mb-2 mt-4"
    >
      Units
    </div>
    <div>
      <span
        class="font-semibold"
      >
        :
      </span>
        available
    </div>
    <div
      class="font-sans font-semibold text-base uppercase text-gray-750 tracking-wider border-0 border-b pb-2 mb-2 mt-4"
    >
      Payments
    </div>
    <div>
      No down payment, but you do need to complete 250 hours of sweat equity. Your monthly payment will be 30% of your income.
    </div>
    <div
      class="mt-4 border-0 border-b pb-2 mb-2"
    >
      You will need to attend an information session.
    </div>
  </div>
</DocumentFragment>
`;
